### DISCUSSION TITLE
Save entire page with resource directory from code

### DISCUSSION THREAD

**Discussion by AndreaMontanari89** [MicrosoftEdge/WebView2Feedback#4007] [Q&A]
Hi, 

Is there a way to download the entire HTML page offline with the directory of the resource by some API?
I know there is a way to hook the start of a download but i want to generate that in automatic.

I would like to have the same result by right-clicking in a random point of the page and select "Save As"

![immagine](https://user-images.githubusercontent.com/70373953/207069550-ad3cf57e-a1b8-4d29-ab3d-c8d1d471be80.png)

The result should be an HTML and a directory containing all the resource.

Thanks

**Comment by nishitha-burman** [comment]
Hi @AndreaMontanari89,

Thank you for the feedback! We are looking into this feature. In the meantime, do any of the workarounds discussed here work for you? https://github.com/MicrosoftEdge/WebView2Feedback/issues/1090
